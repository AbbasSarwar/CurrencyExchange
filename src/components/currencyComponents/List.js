import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { v4 as uuidv4 } from 'uuid';
import styles from '../Styles.module.css';
import { LatestData } from '../../redux/APIs';

const List = () => {
  const Dispatch = useDispatch();
  const image1 = useSelector((state) => state.exchange.image1);
  const image2 = useSelector((state) => state.exchange.image2);

  useEffect(() => {
    Dispatch(LatestData());
  }, [Dispatch]);

  const { convert, latest } = useSelector((state) => state.exchange);
  return (
    <section className="listContainer">
      <div className="equalto">
        <h1>Conversion</h1>
        <div className="changing">
          <div className="imags">
            <img src={image1} alt="" />
            {' '}
            {convert?.query?.from}
            <span className="Amount">{convert?.query?.amount ? `Amount: ${convert.query.amount}` : ''}</span>
          </div>
          <p className={convert?.result ? '' : styles.falseLine}>{convert?.result ? convert.result : 'If you found no value then possible thats are not available to convert'}</p>
          <div className="imags">
            {convert?.query?.to}
            <img src={image2} alt="" />
          </div>
        </div>
      </div>
      <ul className="list">
        <div className="displaytop">
          <h2>Currency Code</h2>
          <h2>$USD Latest Price</h2>
          <h2>Currencies</h2>
        </div>
        {latest && latest.rates && Object.entries(latest.rates).map(([currency, rate]) => (
          <li className="listitem" key={uuidv4()}>
            <p>{currency}</p>
            <p>{rate}</p>
          </li>
        ))}
      </ul>
    </section>
  );
};

export default List;
