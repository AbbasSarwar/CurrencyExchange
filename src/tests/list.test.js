import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import List from './list';

it('renders correctly', () => {
  const mockStore = {
    exchange: {
      image1: 'mock-image1-url',
      image2: 'mock-image2-url',
      convert: {
        query: {
          from: 'USD',
          to: 'EUR',
        },
        result: 1.23,
      },
      latest: {
        rates: {
          EUR: 0.89,
          GBP: 0.73,
          JPY: 110.23,
        },
      },
    },
  };

  const tree = renderer
    .create(
      <BrowserRouter>
        <List exchange={mockStore} />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
