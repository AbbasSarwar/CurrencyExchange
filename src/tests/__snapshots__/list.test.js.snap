// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="listContainer"
>
  <div
    className="equalto"
  >
    <h1>
      Conversion
    </h1>
    <div
      className="changing"
    >
      <div
        className="imags"
      >
        <img
          alt=""
        />
         
      </div>
      <p>
        If you found no value then possible thats are not available to convert
      </p>
      <div
        className="imags"
      >
        <img
          alt=""
        />
      </div>
    </div>
  </div>
  <ul
    className="list"
  >
    <div
      className="displaytop"
    >
      <h2>
        Currency Code
      </h2>
      <h2>
        USD converted
      </h2>
      <h2>
        Currencies
      </h2>
    </div>
  </ul>
</section>
`;
